resolve-deps.cljs

```diff
@@ -28,31 +45,67 @@
           (re-seq #"^\+" dep) {:after (S/replace dep #"^\+" "")}
           :else               dep)))))
 
-(defn load-dep-file-graph
-  "Takes path (a directory path) and dep-file-name (defaults to
-  'deps') and finds all files matching path/*/dep-file-name. Returns
-  a map of directory names to parsed dependencies from the dep file in
-  that directory."
-  [path & [dep-file-name]]
-  (P/let [dep-file-name (or dep-file-name "deps")
-          path-dirs (P/->> (fs/readdir path #js {:withFileTypes true})
-                           (filter #(.isDirectory %))
-                           (map #(.-name %)))
-          dep-list (P/all (for [dname path-dirs
-                                :let [dpath (path/join path dname "deps")]]
-                            (P/catch (P/->> (fs/readFile dpath "utf8")
-                                            parse-dep-str
-                                            (vector dname))
-                              #(vector dname nil))))]
-    (into {} dep-list)))
-
-;; First argument is path to deps directory and the rest are dep
-;; strings. Load all dep definitions under path and then return the
-;; best resolution and order that fulfills the dep strings.
-(P/let [[path & start-dep-strs] *command-line-args*
-        start-dep-str (S/join "," start-dep-strs)
-        file-graph (load-dep-file-graph path)
-        dep-graph (assoc file-graph :START (parse-dep-str start-dep-str))
-        deps (resolve-dep-order dep-graph :START)]
-    (println (S/join " " (filter #(not= :START %) deps))))
+(defn load-deps
+  "Takes a paths sequence and finds all files matching
+  path/*/deps-file. Returns a map of dep nodes to node data. Each node
+  data map contains:
+    - node: name of the node
+    - path: path to dep directory
+    - dep-str: raw dependency string from path/*/deps-file (if it exists)
+    - deps: parsed dependency string"
+  [paths & [deps-file]]
+  (P/let [deps-file (or deps-file "deps")
+          dirdirs (P/all (for [dir paths]
+                           (fs/readdir dir #js {:withFileTypes true})))
+          deps (P/->> (apply concat dirdirs)
+                      (filter #(.isDirectory %))
+                      (map #(P/let [node (.-name %)
+                                    path (path/join (.-path %) node)
```

Comment:
FYI dirent.path is added in v18.17.0 and is now deprecated.  Probably still makes sense to use here, but mentioning in case we hit issues and want to bail and use the user-provided path as the base in the future.

Reply:
Also, it looks like the tests fail.  This line should just be `path (.-path %)`; otherwise, the path to deps-file looks something like 'tests/order/o1/o1/deps'.

Reply:
Good call. I've refactored it to use the original path instead of the `.-path` behavior that is not stable (it works for me in node 21.7).

Reply:
GHA passes now: https://github.com/kanaka/resolve-deps/actions/runs/8649675916